File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/highriskvenuesupload/VenuesUploadResult.java
package uk.nhs.nhsx.highriskvenuesupload;

public class VenuesUploadResult {

    enum ResultType {Ok, ValidationError}

    public final ResultType type;
    public final String message;

    private VenuesUploadResult(ResultType type, String message) {
        this.type = type;
        this.message = message;
    }

    public static VenuesUploadResult ok() {
        return new VenuesUploadResult(ResultType.Ok, "successfully uploaded");
    }

    public static VenuesUploadResult validationError(String message) {
        return new VenuesUploadResult(ResultType.ValidationError, message);
    }

}

File: src/aws/lambdas/incremental_distribution/cta/src/main/java/uk/nhs/nhsx/highriskvenuesupload/VenuesUploadResult.kt
package uk.nhs.nhsx.highriskvenuesupload

sealed class VenuesUploadResult {
    data class Success(val message: String) : VenuesUploadResult()
    data class ValidationError(val message: String) : VenuesUploadResult()
    companion object {
        fun ok(): VenuesUploadResult = Success("successfully uploaded")
        fun validationError(message: String): VenuesUploadResult = ValidationError(message)
    }
}
